float32 theta1
float32 theta2
float32 theta4
float32 b3